examples/Bookshelf/androidApp/src/main/java/io/realm/sample/bookshelf/android/ui/BookshelfViewModel.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.sample.bookshelf.android.ui
+
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import io.realm.sample.bookshelf.BookshelfSDK
+import io.realm.sample.bookshelf.model.Book
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+class BookshelfViewModel : ViewModel() {
+    private var sdk = BookshelfSDK()
+
+    val savedBooks: StateFlow<List<Book>> = sdk.allBooksAsFlowable()
```

Comment:
Would it make sense to try and have the state flow in shared code? Or is that pushing to much Android-style to iOS?

Reply:
StateFlow isn't Android-specific and can be used across different platforms. I think it would be a good idea to move to the BookshelfSDK/repository class - which is what John O'Reilly is doing here: https://johnoreilly.dev/posts/state-flow-multiplatform/

Reply:
I think this approach is still WIP since you need to wrap the Flow with Swift Combine as described in https://johnoreilly.dev/posts/kotlinmultiplatform-swift-combine_publisher-flow/ 
I think `ViewModel` still does makes sense in its respective platform, however, we could add another sample exploring this idea. One goal of this project is to be a template/[Kata](https://en.wikipedia.org/wiki/Kata_(programming)) to express such nuanced implementation details